src/main/java/org/zalando/nakadi/controller/SubscriptionStreamController.java

```diff
@@ -114,26 +138,8 @@ public void onException(final Exception ex) {
             if (!headersSent) {
                 headersSent = true;
                 try {
-                    if (ex instanceof AccessDeniedException) {
-                        writeProblemResponse(response, out, Problem.valueOf(FORBIDDEN,
-                                ((AccessDeniedException) ex).explain()));
-                    } else if (ex instanceof SubscriptionPartitionConflictException) {
-                        writeProblemResponse(response, out, Problem.valueOf(CONFLICT,
-                                ex.getMessage()));
-                    } else if (ex instanceof WrongStreamParametersException) {
-                        writeProblemResponse(response, out, Problem.valueOf(UNPROCESSABLE_ENTITY,
-                                ex.getMessage()));
-                    } else if (ex instanceof NoSuchSubscriptionException) {
-                        writeProblemResponse(response, out, Problem.valueOf(NOT_FOUND,
-                                ex.getMessage()));
-                    } else if (ex instanceof InternalNakadiException) {
-                        writeProblemResponse(response, out, Problem.valueOf(INTERNAL_SERVER_ERROR, ex.getMessage()));
-                    } else if (ex instanceof NoStreamingSlotsAvailable) {
-                        writeProblemResponse(response, out, Problem.valueOf(CONFLICT, ex.getMessage()));
-                    } else {
-                        writeProblemResponse(response, out, Problem.valueOf(SERVICE_UNAVAILABLE,
-                                "Failed to continue streaming"));
-                    }
+                    writeProblemResponse(response, out, exceptionProblem.getOrDefault(ex.getClass(),
```

Comment:
I think there is a problem with this refactoring. 
`instanceof` checks if the class is instance of some class or its subclass. 
The way it is now will only check if it is instance of specific class.
So with this refactoring the behavior can change.

Reply:
ok, all these classes don't have child classes. So it will work fine.